src/components/sequencing/SequenceForm.svelte

```diff
@@ -62,25 +70,73 @@
     pageSubtitle = mode === 'edit' ? savedSequenceName : '';
     saveButtonText = mode === 'edit' && !sequenceModified ? 'Saved' : 'Save';
   }
+  $: {
+    if (sequenceParcelId) {
+      $parcel = $parcels.find(p => p.id === sequenceParcelId) ?? null;
 
-  onMount(() => {
-    if (mode === 'edit') {
-      getUserSequenceSeqJson();
+      loadSequenceAdaptation($parcel?.sequence_adaptation_id);
     }
+  }
+
+  onDestroy(() => {
+    resetSequenceAdaptation();
   });
 
-  async function getUserSequenceFromSeqJson() {
-    const file: File = seqJsonFiles[0];
-    const text = await file.text();
-    const seqJson = JSON.parse(text);
-    const sequence = await effects.getUserSequenceFromSeqJson(seqJson, user);
-    sequenceDefinition = sequence;
-    sequenceSeqJson = text;
+  function resetSequenceAdaptation(): void {
+    globalThis.CONDITIONAL_KEYWORDS = undefined;
+    globalThis.LOOP_KEYWORDS = undefined;
+    globalThis.GLOBALS = undefined;
+    globalThis.ARG_DELEGATOR = undefined;
+    globalThis.LINT = () => undefined;
+    globalThis.TO_SEQ_JSON = () => undefined;
+  }
+
+  async function loadSequenceAdaptation(id: number | null | undefined): Promise<void> {
+    if (id) {
+      const adaptation = await effects.getSequenceAdaptation(id, user);
+
+      if (adaptation) {
+        Function(adaptation.adaptation)();
```

Comment:
Can we add a comment here as to what is potentially being executed?

Reply:
Is this potentially any javascript?

Reply:
Can we put this in a try/catch so it potentially doesn't crash the page?

Reply:
Yeah it's JS, I'll move it into a try catch.